commands/locations/locations.go

```diff
@@ -0,0 +1,225 @@
+	if !colorize {
+		return text
+	}
+	switch operation {
+	case locationCreation:
+		return color.New(color.FgHiGreen, color.Bold).SprintFunc()(text)
+	case locationUpdate:
+		return color.New(color.FgHiYellow, color.Bold).SprintFunc()(text)
+	case locationDeletion:
+		return color.New(color.FgHiRed, color.Bold).SprintFunc()(text)
+	default:
+		return text
+	}
+}
+
+// AddRow adds a row to a table, with text colored according to the operation
+// longTable specifies table with all headers
+func addRow(table tabutil.Table, colorize bool, operation int, lConfig rest.LocationConfiguration) {
+	colNumber := 3
+
+	coloredColumns := make([]interface{}, colNumber)
+	locProps := lConfig.Properties
+	propKeys := locProps.Keys()
+	for i := 0; i < len(propKeys); i++ {
+		propValue := locProps.Get(propKeys[i])
+		value := fmt.Sprintf("%v", propValue)
+		prop := propKeys[i] + ": " + value
+		if i == 0 {
+			coloredColumns[0] = getColoredText(colorize, lConfig.Name, operation)
+			coloredColumns[1] = getColoredText(colorize, lConfig.Type, operation)
+			coloredColumns[2] = getColoredText(colorize, prop, operation)
+		} else {
+			coloredColumns[0] = getColoredText(colorize, "", operation)
+			coloredColumns[1] = getColoredText(colorize, "", operation)
+			coloredColumns[2] = getColoredText(colorize, prop, operation)
+
+		}
+		table.AddRow(coloredColumns...)
+	}
+}
+
+// getLocationConfig makes a GET request to get a given location's configuration
+func getLocationConfig(client httputil.HTTPClient, locationName string) (rest.LocationConfiguration, error) {
+	var locConfig rest.LocationConfiguration
+	request, err := client.NewRequest("GET", getLocationPath(locationName), nil)
+	if err != nil {
+		return locConfig, err
+	}
+	request.Header.Add("Accept", APPJSON)
+	response, err := client.Do(request)
+	if err != nil {
+		return locConfig, err
+	}
+
+	defer response.Body.Close()
+	httputil.HandleHTTPStatusCode(response, locationName, "locations", http.StatusOK)
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return locConfig, err
+	}
+	err = json.Unmarshal(body, &locConfig)
+	if err != nil {
+		return locConfig, err
+	}
+	return locConfig, nil
+}
+
+// putLocationsConfig makes a PUT request to locations API
+func putLocationConfig(client httputil.HTTPClient, locConfig rest.LocationConfiguration) (string, error) {
+
+	locationName := locConfig.Name
+	locationRequest := rest.LocationRequest{
+		Type:       locConfig.Type,
+		Properties: locConfig.Properties,
+	}
+	bArray, err := json.Marshal(locationRequest)
+	request, err := client.NewRequest("PUT", getLocationPath(locationName), bytes.NewBuffer(bArray))
+	if err != nil {
+		return locationName, err
+	}
+	request.Header.Add("Content-Type", APPJSON)
+	response, err := client.Do(request)
+	if err != nil {
+		return locationName, err
+	}
+
+	//httputil.HandleHTTPStatusCode(response, "", "locations", http.StatusCreated)
+	httputil.HandleHTTPStatusCode(response, "", "locations", http.StatusOK)
```

Comment:
The REST endpoint is returning http.StatusCreated, so attempting to add a location fails on error:
```
Error: Expecting HTTP Status code in [200] but got 201, reason "201 Created"
```

Reply:
Fixed